test/unit/app/controllers/network/network-controller-test.js

```diff
@@ -34,8 +34,10 @@ describe('NetworkController', function () {
         assert.equal(providerProxy.test, true)
       })
     })
+
     describe('#getNetworkState', function () {
       it('should return loading when new', function () {
+        networkController = new NetworkController()
```

Comment:
It's more than just timing that differs here. `initializeProvider` hasn't been called on this controller instance.

Any idea where the difference in timing is? It's not clear from reading this test.

Reply:
Ah, answered in the other comment (https://github.com/MetaMask/metamask-extension/pull/9088#discussion_r461202264)

Reply:
On newly initialized network controllers, `initializeProvider` eventually updates the network state from `loading` to whatever the selected chain is (in these tests, `4`). From the test description, I believe it's trying to test whether the initial state of the controller is in fact `loading`, and we don't need to call `initializeProvider` to test that.

As mentioned elsewhere, I fixed it so that `intializeProvider` is only called where necessary in: b3e5bef